lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1390,6 +1392,106 @@ extension DomCSSRuleListExtension on DomCSSRuleList {
+@JS()
+@staticInterop
+abstract class DomTrustedTypePolicyFactory {}
+
+/// (Some) methods of the [DomTrustedTypePolicyFactory]:
+extension DomTrustedTypePolicyFactoryExtension on DomTrustedTypePolicyFactory {
+  /// createPolicy
+  external DomTrustedTypePolicy createPolicy(
+    String policyName,
+    DomTrustedTypePolicyOptions? policyOptions,
+  );
+}
+
+/// Options to create a trusted type policy.
+@JS()
+@staticInterop
+@anonymous
+abstract class DomTrustedTypePolicyOptions {
+  /// Constructs a TrustedPolicyOptions object in JavaScript.
+  ///
+  /// The following properties need to be manually wrapped in [allowInterop]
+  /// before being passed to this constructor: [createScriptURL].
+  external factory DomTrustedTypePolicyOptions({
+    DomCreateScriptUrlOptionFn? createScriptURL,
+  });
+}
+
+/// Type of the function to configure createScriptURL
+typedef DomCreateScriptUrlOptionFn = String? Function(String input);
+
+/// An instance of a TrustedTypePolicy
+@JS()
+@staticInterop
+abstract class DomTrustedTypePolicy {}
+
+/// (Some) methods of the [DomTrustedTypePolicy]
+extension DomTrustedTypePolicyExtension on DomTrustedTypePolicy {
+  /// Create a `TrustedScriptURL` for the given [input].
+  external DomTrustedScriptUrl createScriptURL(String input);
+}
+
+/// An instance of a DomTrustedScriptUrl
+@JS()
+@staticInterop
+abstract class DomTrustedScriptUrl {}
+
+/// (Some) methods of the [DomTrustedTypePolicy]
+extension DomTrustedScriptUrlExtension on DomTrustedScriptUrl {
+  String get url => js_util.callMethod<String>(this, 'toString', <String>[]);
+}
+
+// The expected set of files that the flutter-engine TrustedType policy is going
+// to accept as valid.
+const Set<String> _expectedFilesForTT = <String>{
+  'canvaskit.js',
+};
+
+// The definition of the `flutter-engine` TrustedType policy.
+DomTrustedTypePolicy get _ttPolicy {
+  return domWindow.trustedTypes!.createPolicy('flutter-engine',
+    DomTrustedTypePolicyOptions(
+      // Validates the given [url].
+      createScriptURL: allowInterop((String url) {
+        final Uri uri = Uri.parse(url);
+        if (_expectedFilesForTT.contains(uri.pathSegments.last)) {
+          return uri.toString();
+        }
+        domWindow.console.error(
+          'URL rejected by TrustedTypes policy flutter-engine: $url'
+          '(download prevented)');
+        return null; // Ends up in JS as an empty string
+      }
+    )));
+}
+
+/// Converts a String [url] into a [DomTrustedScriptUrl] object (if available).
+Object createTrustedScriptUrl(String url) {
+  if (domWindow.trustedTypes != null) {
+    // Pass `url` through Flutter Engine's TrustedType policy.
+    final DomTrustedScriptUrl trustedCanvasKitUrl =
+      _ttPolicy.createScriptURL(url);
+
+    assert(trustedCanvasKitUrl.url != '',
+      'URL: $url rejected by TrustedTypePolicy');
+
+    return trustedCanvasKitUrl;
+  }
+  return url;
+}
+
+// Utility methods below
```

Comment:
In my experience comments that designate a category of top-level members get out of date quickly because an evolution of a file, especially a long one like this, is a lot of small changes with authors not re-reading the whole file to know where to put their code.

Reply:
Removed both "bookend" comments.